docs/src/internals/ir.md

```diff
@@ -28,5 +36,45 @@ Block {
 
 Instruction {
     opcode: u8
+    num_operands: u32
+    operands: Operand[num_operands]
+}
+
+Constant {
+    type_index: usize,
+    num_bytes: usize,
+    bytes: u8[num_bytes]
+}
+```
+
+
+### Type encoding.
+
+`Type` is encoded as one of the following:
+
+IntegerType {
+    type_kind: u8 = 0
+    num_bits: u32
```

Comment:
Isn't `num_bits` implied by `type_kind`?

Reply:
`Type` can only express the size of a type in bytes, whereas this is a measure of bits. I may be able to fix that. Bear with me.

Reply:
Hold on, I misunderstood.

type_kind 0 just says "it's an integer". It's up to every type kind to communicate its size. Is that OK, or should we have a top-level struct that indicates the size in a generic fashion, then a sub-enum for the type-specific parts?

Maybe we can kick this down the road a bit until we implement some more types?

Reply:
I see what you mean about allowing arbitrary bit sizes. OK for me to worry about more details later.